sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_discovery.py

```diff
@@ -0,0 +1,104 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+import os
+import time
+from dataclasses import dataclass
+import dns.resolver
+from dns.resolver import NXDOMAIN, LifetimeTimeout, NoNameservers  # cspell:disable-line
+from dns.rdatatype import SRV  # cspell:disable-line
+from ._constants import DISABLE_APPCONFIGURATION_DISCOVERY, HTTPS_PREFIX
+
+
+@dataclass
```

Comment:
Why is this a `dataclass`?

Reply:
I'm not sure what you mean it's the `@dataclass` that represents an SRV record. The rest of the module is outside of this class.

Reply:
Why does it have to be a data class and not just a class? We are redefining the __init__ anyways. What data class functionality are we using?

Reply:
removed